packages/salesforcedx-vscode-soql/src/client/client.ts

```diff
@@ -55,9 +70,29 @@ export function startLanguageClient(context: ExtensionContext): void {
     serverOptions,
     clientOptions
   );
+  client.registerFeature(new QueryValidationFeature());
 
   // Start the client. This will also launch the server
   client.start();
+
+  await client.onReady().then(() => {
+    client.onRequest(RequestTypes.RunQuery, async (queryText: string) => {
+      try {
+        const conn = await workspaceContext.getConnection();
+        const queryData = await new QueryRunner(conn).runQuery(queryText, {
+          showErrors: false
+        });
+        return { result: queryData };
+      } catch (e) {
+        const error = {
+          name: e.name,
+          errorCode: e.errorCode,
+          message: e.message
+        };
+        return { error };
+      }
+    });
+  });
```

Comment:
Might be able to make this a little more readable:

```typescript
await client.onReady()
client.onRequest(RequestTypes.RunQuery, async (queryText: string) => {
  try {
    const conn = await workspaceContext.getConnection();
    const queryData = await new QueryRunner(conn).runQuery(queryText, {
      showErrors: false
    });
    return { result: queryData };
  } catch (e) {
    // simplify, unless you're specifically trying to omit other properties
    return { error: e };
  }
});
```

Not sure if you want to await client.onRequest or not, it seems more like a subscribe call but could be wrong.

Reply:
Thanks. I'll change it to use `await`.
The caught exception cannot be returned as-is because the return value has to be a plain JS object (to be serialized as a JSON-RPC response). I'll add a comment to make it clear.